Add a subsessionTimestamp to the executive summary

trink opened the pull request.

trink committed:
Add a subsessionTimestamp to the executive summary

trink commented on heka/sandbox/decoders/extract_executive_summary.lua:
```diff
@@ -232,6 +237,20 @@ function process_message()
             end
+
+            if type(info.subsessionStartDate) == "string" then
+                local t = dt.rfc3339:match(info.subsessionStartDate)
```
This timestamp is a little screwed up (some values aren't ISO as spec'ed). I am thinking we may just want the date part (treat it UTC and ignore the timezone). i.e. just keep the else clause.

`2015-08-05T00:00:00.0-07:00`
`2015-08-05T00:00:00`

mreid-moz commented on heka/sandbox/decoders/extract_executive_summary.lua:
```diff
@@ -232,6 +237,20 @@ function process_message()
+                    if t then
+                        msg.Fields.subsessionTimestamp.value = dt.time_to_ns(t)
+                    else
+                        msg.Fields.subsessionTimestamp.value = cts
```
I don't think we should fall back to creationTimestamp here - we already have that value in a separate field, so we should just leave out the subsessionTimestamp if it's unavailable (or unparseable).

trink replied:
The filtering and computation work off of subsessionTimestamp.  If it is not set the code will have to fall back to creationTimestamp anyway.  I can remove it from here but would have to push it out to any consumers of the data.

Edit: would still have to special case crash pings unless I use the crash date for the subsessionTime

mreid-moz replied:
Yeah, I'd rather let consumers decide what to do about it.

trink commented:
Cleaning up the executive_summary as discussed in IRC
- removing un-used sessionId and subsessionCounter
- standardizing on an activityTimestamp

trink committed:
Clean up the executive_summary as discussed in IRC

- remove un-used sessionId and subsessionCounter
- standardize on an activityTimestamp

trink committed:
Fix the crashDate extraction

mreid-moz commented on heka/sandbox/decoders/extract_executive_summary.lua:
```diff
@@ -211,26 +212,28 @@ function process_message()
-        msg.Fields.sessionId.value          = ""
-        msg.Fields.subsessionCounter.value  = 0
-        msg.Fields.pluginHangs.value        = 0
+        msg.Fields.hours.value               = 0
```
nit: some of this alignment whitespace can be removed (also above in "crash_fields" and "main_fields").

trink committed:
Remove the extra alignment stop (whitespace)
